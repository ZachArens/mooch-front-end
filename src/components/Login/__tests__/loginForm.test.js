import React from 'react';
import {cleanup, fireEvent, render} from '@testing-library/react';
import LoginForm from '../loginForm';

const updateFields = jest.fn();

beforeEach(() => {
    cleanup();
})

describe('<LoginForm />', () => {
    
    test('renders without crashing', () => {
        render(<LoginForm emailName="1234" passwordName="5678" updateFields={updateFields} />);
    });

    test('calls updateFields on change of both fields', () => {
        const emailName = "test1";
        const passwordName = "test2";

        const {queryByTestId } = render(<LoginForm emailName={emailName} passwordName={passwordName} 
            updateFields={updateFields} />);

            fireEvent.change(queryByTestId(emailName), {
                target: {defaultValue: "test@testemail.com"},
            });
    
            fireEvent.change(queryByTestId(passwordName), {
                target: {defaultValue: "1234asdfas"},
            });

            expect(updateFields).toHaveBeenCalledTimes(2);
    });  

    test('matches snapshot', () => {
        const { container } = render(<LoginForm />);
        expect(container.firstChild).toMatchSnapshot();
    });
});