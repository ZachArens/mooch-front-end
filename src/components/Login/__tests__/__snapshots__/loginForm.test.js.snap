// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> matches snapshot 1`] = `
<div>
  <label
    data-testid="emailLabel"
  >
    Email
    <input
      placeholder="email"
      type="text"
    />
  </label>
  <label
    data-testid="passwordLabel"
  >
    Password
    <input
      placeholder="********"
      type="password"
    />
  </label>
</div>
`;
